### PR TITLE
telemetry(amazonq): add telemetry for code generated and code accepted

### PR DESCRIPTION
**Pull request by kelvin-klchu**
## Problem
- we need to define a way to measure code generated and code accepted from `/dev` feature

## Solution
- we compute the char and line difference between the two files
- all generated codes are immediately reported in telemetry
- within the same session, de-duplication is done to avoid reporting the exact same changes twice across different code gen iterations
- currently, only accepted changes with new lines added will be reported

## Special Note
`charsAdded` and `charRemoved` may be slightly different from the count you manually get from the IDE.
The reason is because new line character `\n` is not counted here in the computeDiff method.

---

License: I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: packages/core/src/amazonqFeatureDev/session/session.ts [hunk 1]
```diff
@@ -209,6 +212,7 @@ export class Session {
     }
 
     public async insertNewFiles(newFilePaths: NewFileInfo[]) {
```
Comment (neilk-aws) [inline packages/core/src/amazonqFeatureDev/session/session.ts:214]: If a customer applies files from the file tree, but never clicks "Accept remaining", do we count anything in this approach?

**Review by neilk-aws — APPROVED** [review]

File: packages/core/src/amazonqFeatureDev/session/session.ts (hunk 1 above)
Comment (kelvin-klchu) [reply packages/core/src/amazonqFeatureDev/session/session.ts:214]: Yes. This method is called by both the accept changes button and also by the individual file accept action button.

File: packages/core/package.json [hunk 2]
```diff
@@ -488,6 +488,7 @@
         "bytes": "^3.1.2",
         "cross-fetch": "^4.0.0",
         "cross-spawn": "^7.0.3",
+        "diff": "^7.0.0",
```
Comment (jpinkney-aws) [inline packages/core/package.json:491]: we technically have this as a dev dependency right now btw. If we want to promoted it to a regular dependency we should remove it from there

**Comment by kelvin-klchu** [comment]
rebase changes and resolved conflicts

File: packages/core/package.json (hunk 2 above)
Comment (kelvin-klchu) [reply packages/core/package.json]: I have moved this to dependencies, also keep the version as `^5.1.0` to avoid breaking existing functionalities.

**Review by jpinkney-aws — APPROVED** [review]

**Pull request merged**